c++/src/ColumnWriter.cc

```diff
@@ -1103,6 +1115,12 @@ namespace orc {
       throw InvalidArgument("Failed to cast to StringVectorBatch");
     }
 
+    EncodedStringVectorBatch* encoded_string_batch =
+        dynamic_cast<EncodedStringVectorBatch*>(&rowBatch);
+    if (encoded_string_batch) {
```

Comment:
Maybe let the caller call calculateDataAndLength() is better?

Reply:
I'm afrait it will increase user's mental burden. Do you think make it transparent for user is ok?